### PR TITLE
Rubocop lint fixes

### PR DESCRIPTION
**Pull request by joshlory**
I've aligned with these enforced styles since they seem to be more popular in the codebase:

``` yml
Style/AlignParameters:
  EnforcedStyle: with_fixed_indentation

Style/FirstParameterIndentation:
  EnforcedStyle: consistent
```

### CODE REVIEW

**Comment by ashercodeorg** [comment]
In both cases, do we want to set `IndentationWidth` to a non-default value? Looking at the diffs, it seems many places use `+4` rather than `+2` for it. Being able to visually tell "why" the line is continuing seems useful, at least to me.

For documentation, see https://github.com/bbatsov/rubocop/blob/master/config/default.yml#L178-L180 and https://github.com/bbatsov/rubocop/blob/master/config/default.yml#L178-L180.

File: bin/i18n-codeorg/lib/merge-all-locales.rb [hunk 1]
```diff
@@ -24,8 +24,8 @@ def merge_translation_tree(en_translation, new_translation, prev_translation)
       if en_translation.has_key?(key) && prev_translation.has_key?(key)
         new_translation[key] =
           merge_translation_tree(en_translation[key],
-                                 new_translation[key],
-                                 prev_translation[key])
+            new_translation[key],
+            prev_translation[key])
```
Comment (ashercodeorg) [inline bin/i18n-codeorg/lib/merge-all-locales.rb]: We should adopt a standard as to where the closing parenthesis goes. Do you prefer

```
new_translation[key] =
  merge_translation_tree(en_translation[key],
    new_translation[key],
    prev_translation[key])
```

or

```
new_translation[key] =
  merge_translation_tree(en_translation[key],
    new_translation[key],
    prev_translation[key]
  )
```

Of course, this is for another PR.

File: dashboard/app/models/net_sim.rb [hunk 2]
```diff
@@ -95,9 +95,9 @@ def self.message_granularity_options
 
   def self.create_from_level_builder(params, level_params)
     create!(level_params.merge(
-              user: params[:user],
-              game: Game.netsim,
-              level_num: 'custom'
+      user: params[:user],
+      game: Game.netsim,
+      level_num: 'custom'
             ))
```
Comment (ashercodeorg) [inline dashboard/app/models/net_sim.rb]: These parens could (should) be de-indented.

File: dashboard/lib/sqs/queue_processor.rb [hunk 3]
```diff
@@ -92,7 +92,7 @@ def start
                 # pause even if the handler failed because we don't want to exceed the configured rate
                 # even when failures are occuring.
                 delay = rate_limiter.inter_batch_delay(
-                    batch_size: batch_size, elapsed_time_sec: Time.now.to_f - start_time_sec)
+                  batch_size: batch_size, elapsed_time_sec: Time.now.to_f - start_time_sec)
```
Comment (ashercodeorg) [inline dashboard/lib/sqs/queue_processor.rb]: Break the `elapsed_time_sec` into another line?

File: dashboard/test/controllers/activities_controller_test.rb [hunk 4]
```diff
@@ -145,7 +145,7 @@ def _test_logged_in_milestone(async_activity_writes:)
                       user_agent: 'Rails Testing',
                       locale: :'en-us'
                   }],
-                 slogger.records)
+      slogger.records)
```
Comment (ashercodeorg) [inline dashboard/test/controllers/activities_controller_test.rb]: This looks funny, given the indentation of the hash in the previous lines.

File: dashboard/test/controllers/api_controller_test.rb [hunk 5]
```diff
@@ -291,7 +291,7 @@ def make_progress_in_section(script)
                       user_agent: 'Rails Testing',
                       locale: :'en-us'
                   }],
-                 slogger.records)
+      slogger.records)
```
Comment (ashercodeorg) [inline dashboard/test/controllers/api_controller_test.rb]: Again (funny, giving previous hash). Also twenty-five lines above.

File: dashboard/test/controllers/omniauth_callbacks_controller_test.rb [hunk 6]
```diff
@@ -45,14 +45,14 @@ class OmniauthCallbacksControllerTest < ActionController::TestCase
...
-                info: stub('info', nickname: '',
-                           name: {'first' => 'Hat', 'last' => 'Cat'},
-                           email: 'first_last@clever_teacher.xx',
-                           user_type: 'teacher',
-                           dob: nil,
-                           gender: nil))
+      uid: '1111',
+      provider: 'clever',
+      info: stub('info', nickname: '',
+                 name: {'first' => 'Hat', 'last' => 'Cat'},
```
Comment (ashercodeorg) [inline dashboard/test/controllers/omniauth_callbacks_controller_test.rb]: Should the `stub` parameters also be indented `+2`? Also below.

File: dashboard/test/ui/support/cookie_helpers.rb [hunk 7]
```diff
@@ -18,13 +18,13 @@ def decrypt_cookie(cookies)
     message = CGI.unescape(session_cookie[:value])
 
     key_generator = ActiveSupport::KeyGenerator.new(
-                                                    CDO.dashboard_secret_key_base,
-                                                    iterations: 1000
+      CDO.dashboard_secret_key_base,
+      iterations: 1000
                                                     )
```
Comment (ashercodeorg) [inline dashboard/test/ui/support/cookie_helpers.rb]: De-indent the closing paren. Also below.

File: lib/cdo/hip_chat.rb [hunk 8]
```diff
@@ -113,13 +113,13 @@ def self.post_hipchat_form(room, message, options)
...
-                                                 room_id: room.to_s,
-                                                 message: body
-                                             }))
+      uri,
+      {color: 'gray'}.merge(options).merge({
+                                               from: @@name,
+                                               auth_token: @@auth_token,
+                                               room_id: room.to_s,
+                                               message: body
+                                           }))
```
Comment (ashercodeorg) [inline lib/cdo/hip_chat.rb]: De-indent.

File: pegasus/forms/professional_development_workshop.rb [hunk 9]
```diff
@@ -36,9 +36,9 @@ def self.progress_snapshot(section_id)
     DASHBOARD_DB[:followers].
       join(:users, id: :student_user_id).
       select(Sequel.as(:student_user_id, :id),
-             :users__id___id,
-             :users__name___name,
-             :users__email___email,
+        :users__id___id,
+        :users__name___name,
+        :users__email___email,
             ).
```
Comment (ashercodeorg) [inline pegasus/forms/professional_development_workshop.rb]: De-indent the closing paren.

File: pegasus/helpers/section_api_helpers.rb [hunk 10]
```diff
@@ -58,8 +58,8 @@ def self.fetch_if_allowed(id_or_ids, dashboard_user_id)
       where(users__id: id, users__deleted_at: nil).
       left_outer_join(:secret_pictures, id: :secret_picture_id).
       select(*fields,
-             :secret_pictures__name___secret_picture_name,
-             :secret_pictures__path___secret_picture_path,
+        :secret_pictures__name___secret_picture_name,
+        :secret_pictures__path___secret_picture_path,
             ).
```
Comment (ashercodeorg) [inline pegasus/helpers/section_api_helpers.rb]: De-indent the closing paren.

File: pegasus/test/test_dashboard.rb [hunk 11]
```diff
@@ -35,15 +35,15 @@ class DashboardTest < Minitest::Test
                           name: FakeDashboard::STUDENT[:name],
                           admin: FakeDashboard::STUDENT[:admin]
                      },
-                     @student.select(:name, :admin))
+          @student.select(:name, :admin))
```
Comment (ashercodeorg) [inline pegasus/test/test_dashboard.rb]: De-indent the hash. Also below.

File: pegasus/test/test_v2_user_routes.rb [hunk 12]
```diff
@@ -31,7 +31,7 @@ class V2UserRoutesTest < Minitest::Test
                          'name' => FakeDashboard::STUDENT[:name],
                          'owned_sections' => []
                      },
-                     JSON.parse(@pegasus.last_response.body))
+          JSON.parse(@pegasus.last_response.body))
```
Comment (ashercodeorg) [inline pegasus/test/test_v2_user_routes.rb]: De-indent the hash. Also below.

**Comment by ashercodeorg** [comment]
Generally LGTM, though I think there should be some consideration as to whether the indent should be `+2` or `+4`, as well as some cleanup of the (presumably) auto-generated corrections.

File: bin/i18n-codeorg/lib/merge-all-locales.rb (hunk 1 above)
Comment (joshlory) [inline bin/i18n-codeorg/lib/merge-all-locales.rb]: Slightly prefer this:

``` ruby
new_translation[key] = merge_translation_tree(
  en_translation[key],
  new_translation[key],
  prev_translation[key]
)
```

Weirdly I can't find the Rubocop rule to enforce newlines after/before parens for multiline params.

File: dashboard/app/models/net_sim.rb (hunk 2 above)
Comment (joshlory) [inline dashboard/app/models/net_sim.rb]: Yep!  Rubocop just merged a fix for the `Style/ClosingParenthesisIndentation` cop: https://github.com/bbatsov/rubocop/issues/1758

I can hand-fix occurrences in this PR, then enable the cop once it ships.

File: dashboard/test/controllers/omniauth_callbacks_controller_test.rb (hunk 6 above)
Comment (joshlory) [inline dashboard/test/controllers/omniauth_callbacks_controller_test.rb]: Yeah, it should.  Maybe Rubocop is getting confused with the nested method calls?  I'll fix by hand.

**Comment by joshlory** [comment]
I think the most common Ruby style is [two spaces for all indents](https://github.com/bbatsov/ruby-style-guide#spaces-indentation), including line continuations.

I'm happy to setup Rubocop to enforce either +2 or +4 as long as we're consistent.  Maybe a slight preference for +2 for compactness.

File: pegasus/test/test_v2_user_routes.rb (hunk 12 above)
Comment (joshlory) [inline pegasus/test/test_v2_user_routes.rb]: I've been updating these to

``` ruby
assert_equal({
  'id' => FakeDashboard::STUDENT[:id],
  'admin' => FakeDashboard::STUDENT[:admin],
  'name' => FakeDashboard::STUDENT[:name],
  'owned_sections' => []
}, JSON.parse(@pegasus.last_response.body))
```

But I wonder if it's better to write it as

``` ruby
assert_equal(
  {
    'id' => FakeDashboard::STUDENT[:id],
    'admin' => FakeDashboard::STUDENT[:admin],
    'name' => FakeDashboard::STUDENT[:name],
    'owned_sections' => []
  },
  JSON.parse(@pegasus.last_response.body)
)
```

File: bin/i18n-codeorg/lib/merge-all-locales.rb (hunk 1 above)
Comment (ashercodeorg) [inline bin/i18n-codeorg/lib/merge-all-locales.rb]: Agreed. And I couldn't find it either (though I didn't try much).

File: pegasus/test/test_v2_user_routes.rb (hunk 12 above)
Comment (ashercodeorg) [inline pegasus/test/test_v2_user_routes.rb]: The latter looks more readable to me.

**Comment by ashercodeorg** [comment]
Let's do +2, then (despite my personal preference for +4).

**Comment by joshlory** [comment]
Addressed your feedback in e79c5e9f1a110925880b0ae7e4f47e3c67eb38af.  Let me know what you think!

File: shared/test/test_net_sim_api.rb [hunk 13]
```diff
@@ -473,32 +473,47 @@ def many_node_delete_cascading_generates_minimum_invalidations
...
+      channel: @shard_id,
+      event: TABLE_NAMES[:node],
+      data: {
+        action: 'delete',
+        ids: [
+          node_a['id'],
+          node_b['id']
+        ]
+      }
+    })
```
Comment (joshlory) [inline shared/test/test_net_sim_api.rb:516]: Should this be written as the following?

``` ruby
assert_equal(
  test_spy.publish_history[2],
  {
    channel: @shard_id,
    event: TABLE_NAMES[:node],
    data: {
      action: 'delete',
      ids: [
        node_a['id'],
        node_b['id']
      ]
    }
  }
)
```

As it written it looks a little cleaner to me, but I'm biased towards JavaScript-style params:

``` javavscript
f(a, b, c, {
  opt1: true,
  opt2: false
});
```

Comment (ashercodeorg) [inline shared/test/test_net_sim_api.rb:516]: The version in the comment would be my preference, being biased towards C++ style.

**Comment by ashercodeorg** [comment]
LGTM! Thanks for doing this work!

**Comment by ashercodeorg** [comment]
NOTE: This is opened against `staging`, not `staging-next`.

**Pull request merged**